telluric/collections.py

```diff
@@ -218,6 +226,58 @@ def save(self, filename, driver=None):
                 new_feature = self._adapt_feature_before_write(feature)
                 sink.write(new_feature.to_record(crs))
 
+    rasters_executor = ThreadPoolExecutor(max_workers=MAX_WORKERS)
```

Comment:
I understand where does this come from, but I'm not quite sure about adding this property and method here. What do you think of subclassing the `BaseCollection`, or creating some mixin class that can be combined with `FileCollection` or `FeatureCollection`?